### PR TITLE
[exe-unit] Single process killing

### PR DESCRIPTION
**Pull request by boryspoplawski**
These changes are needed to run Yagna in Graphene-SGX, which currently does not fully support `/proc` subsystem (used by `ProcessTree`).

### CODE REVIEW

File: exe-unit/src/runtime/process.rs [hunk 1]
```diff
@@ -164,12 +164,11 @@ impl RuntimeProcess {
                 .stderr(Stdio::piped())
                 .spawn()?;
 
-            let tree =
-                ProcessTree::try_new(child.id()).map_err(|e| Error::RuntimeError(e.to_string()))?;
+            let signle_child = child.id();
```
Comment (mfranciszkiewicz) [inline exe-unit/src/runtime/process.rs:167]: typo in `single_child`

**Review by mfranciszkiewicz — CHANGES_REQUESTED** [review]
Please don't remove `ChildProcess::Tree` and make it feature-enabled in the following fashion:

```rust
enum ChildProcess {
    #[cfg(feature = "sgx")]
    Single { pid: u32 },
    #[cfg(not(feature = "sgx"))]
    Tree(process::ProcessTree),		     
    Service(ProcessService)
}
```

or

```rust
enum ChildProcess {
    #[cfg(feature = "sgx")]
    Subprocess { pid: u32 },
    #[cfg(not(feature = "sgx"))]
    Subprocess { tree: process::ProcessTree },		     
    Service(ProcessService)
}
```

**Comment by boryspoplawski** [comment]
> Please don't remove `ChildProcess::Tree` and [...]

What's the use of `ProcessTree` here? Seems that it always contains a single process anyway ...

File: exe-unit/src/runtime/process.rs (hunk 1 above)
Comment (boryspoplawski) [reply exe-unit/src/runtime/process.rs]: Done.

**Comment by mfranciszkiewicz** [comment]
> > Please don't remove `ChildProcess::Tree` and [...]
> 
> What's the use of `ProcessTree` here? Seems that it always contains a single process anyway ...

We're not limiting runtimes to a single process. Replacing `ProcessTree` with `Single` is like reverting to a pre-poc implementation, which shouldn't be done on a whim

**Comment by boryspoplawski** [comment]
> We're not limiting runtimes to a single process. Replacing `ProcessTree` with `Single` is like reverting to a pre-poc implementation, which shouldn't be done on a whim

I missed that `ProcessTree` iterates through all system processes.
Done.

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**